### ISSUE TITLE
Poll: What is preventing you from adopting ktlint?

### ISSUE THREAD

**Issue by shyiko** [pinterest/ktlint#45]
Please use :+1: to upvote answers below (or add a new comment(s) if you have a different reason(s)).

**Comment by shyiko** [comment]
4 spaces for indentation (please take a vote on #43)

**Comment by shyiko** [comment]
No wildcard imports.

**Comment by shyiko** [comment]
No configuration.

**Comment by rubenwardy** [comment]
I like all the rules, except allowing vertical alignment for assignments. It makes the code much more readable. A rule could be only one space before assignment `=` unless it's lining up with another assignment.

I guess that this may be done for the simplicity of the linter, however.

```kotlin
var asknakdf = 3
var kass     = 4

# versus
var asknakdf = 3
var kass = 4
```

**Comment by gladed** [comment]
OK, not preventing me, but it needs a gradle plugin badly.

**Comment by shyiko** [comment]
@gladed have you seen
- [jlleitschuh/ktlint-gradle](https://github.com/jlleitschuh/ktlint-gradle)
- [jeremymailen/kotlinter-gradle](https://github.com/jeremymailen/kotlinter-gradle)
- [diffplug/spotless](https://github.com/diffplug/spotless/tree/master/plugin-gradle#applying-ktlint-to-kotlin-files)

? (all mentioned in Getting Started :: Gradle at https://ktlint.github.io/)

**Comment by gladed** [comment]
My bad -- didn't see these. Thanks for the references!

**Comment by adam-beneschan** [comment]
Please don't mandate only one space before an end-of-line comment.  We need the ability to line up comments and make them pretty.

**Comment by shyiko** [comment]
@adam-beneschan if you mean [this](https://github.com/shyiko/ktlint/blob/master/ktlint-ruleset-standard/src/test/kotlin/com/github/shyiko/ktlint/ruleset/standard/NoMultipleSpacesRuleTest.kt#L17) then it's not forbidden.

**Comment by adam-beneschan** [comment]
@shyiko Interesting--that's the same sort of thing I was having trouble with (the extra spaces before //; mine wasn't on a when/case, though).  But I'll look into it further.  It's possible that the in-house tool I used is out of date or was modified by someone else to add this behavior.

**Comment by mhf-air** [comment]
It's a little bit slow. I use ktlint in vim to automatically format the code every time I save the file, the problem is that it pauses half a second

**Comment by WhiteHatArpit** [comment]
This one #86

**Comment by panzerfahrer** [comment]
I'd like to stress and upvote the "gradle plugin" topic.

I tried `ktlint-gradle` and `kotlinter-gradle`. Both are really basic and don't offer much more over what I get by hacking up one to three custom gradle tasks.

I think a first-class citizen gradle plugin would be much more appreciated than just yet another one-man-show plugin (no offense meant).

Or maybe a somehow "officially" supported plugin (out of the already existing ones) could help concentrate efforts.

**Comment by jeremymailen** [comment]
@panzerfahrer what are some features on your wish list for a gradle plugin?

Features the come to mind for me are:
- Auto-detection of other plugins and integrating with their build lifecycle (e.g. `check` dependency for kotlin-jvm and android)
- Auto-detection of kotlin sourceSets and creation of tasks and output files for each
- Incremental build support (i.e. tasks are up-to-date until source files change)
- Top-level extension to configure settings for all tasks
- Output format supporting other quality tools (e.g. junit xml report)
- Customization of ktlint version and custom rules

https://github.com/jeremymailen/kotlinter-gradle does support these today. I like that idea that a gradle plugin should be _simple_ to _apply_. It's not much different than the `checkstyle` plugin. Maybe there are some other knobs and dials which are useful?

I'm quite open to contributions and adding other maintainers to the project. Thanks for the input!

**Comment by erikcaffrey** [comment]
Is there a fix to wildcard imports?

**Comment by ChrisZou** [comment]
Not enough built-in rule. It require quite a bit knowledge and effort to create custom rules. Would be  great if all the AndroidStudio code inspection rules are available as ktlint built-in rules.

**Comment by stkent** [comment]
We just added ktlint support to [Gnag](https://github.com/btkelly/gnag/), a Gradle plugin that aims to simplify checking for and reporting code issues. Might be an option for those of you looking for a more solid Gradle plugin that wraps ktlint :)

**Comment by asarazan** [comment]
I would just want turn-key support for all of Google's kotlin style guide

**Comment by elenigen** [comment]
Customizable list of ktlint rules (or at least to be able to turn on & off rules)

Let's be honest, some rules are so annoying it's ruining this tool, just because we cannot turn off it or configure it.

**Comment by lwasyl** [comment]
Ability to disable all built-in rules, and instead run only own, custom rulesets.

**Comment by Zhuinden** [comment]
Disable certain rules. The "no wildcard imports" being non-configurable is a dealbreaker for me.

But I'd also like to configure this particular rule: `Consistent spacing after keywords, commas; around colons, curly braces, parens, infix operators, comments, etc;` so that I can make it allow `if(blah)` instead of `if (blah)` for my own projects.

**Comment by RodrigoAmaroEng** [comment]
Ability to validate XML and other resource files
